website/src/components/Udels/UdelsEditor.jsx

```diff
@@ -0,0 +1,34 @@
+import { ReplContext } from '@src/repl/util.mjs';
+
+import Loader from '@src/repl/components/Loader';
+import { Panel } from '@src/repl/components/panel/Panel';
+import { Code } from '@src/repl/components/Code';
+import BigPlayButton from '@src/repl/components/BigPlayButton';
+import UserFacingErrorMessage from '@src/repl/components/UserFacingErrorMessage';
+
+// type Props = {
+//  context: replcontext,
+//  containerRef:  React.MutableRefObject<HTMLElement | null>,
+//  editorRef:  React.MutableRefObject<HTMLElement | null>,
+//  error: Error
+//  init: () => void
+// }
+
+export default function UdelsEditor(Props) {
+  const { context, containerRef, editorRef, error, init } = Props;
+  const { pending, started, handleTogglePlay } = context;
+  return (
+    <ReplContext.Provider value={context}>
```

Comment:
maybe it would be simpler to pull out this Provider to Repl.jsx and throw in the stuff needed? Then useContext could be used anywhere without "prop drilling". I tend to get confused when props are passed more than 2 levels deep, but I won't say I haven't done it myself :P